repos/system_upgrade/el7toel8/actors/repositoriesblacklist/tests/test_repositoriesblacklist.py

```diff
@@ -1,75 +1,119 @@
-    assert 'codeready-builder-for-rhel-8-x86_64-rpms' in repoids
-
-
-def test_repositoriesblacklist_no_optional(current_actor_context):
-    repo_facts = get_repo_facts(['rhel-7-server-rpms', 'rhel-7-extras-rpms'])
-    current_actor_context.feed(repo_facts)
-    current_actor_context.run()
-    assert current_actor_context.consume(RepositoriesBlacklisted)
-    repoids = current_actor_context.consume(RepositoriesBlacklisted)[0].repoids
-    assert 'codeready-builder-for-rhel-8-x86_64-rpms' in repoids
-
-
-def test_repositoriesblacklist_disabled_optional(current_actor_context):
-    repo_facts = get_repo_facts(['rhel-7-server-optional-rpms'], False)
-    current_actor_context.feed(repo_facts)
-    current_actor_context.run()
-    assert current_actor_context.consume(RepositoriesBlacklisted)
-    repoids = current_actor_context.consume(RepositoriesBlacklisted)[0].repoids
-    assert 'codeready-builder-for-rhel-8-x86_64-rpms' in repoids
-
-
-def test_repositoriesblacklist_optional(current_actor_context):
-    repo_facts = get_repo_facts(['test', 'rhel-7-server-optional-rpms', 'rhel-7-server-rpms'])
-    current_actor_context.feed(repo_facts)
-    current_actor_context.run()
-    assert not current_actor_context.consume(RepositoriesBlacklisted)
-
-
-def test_repositoriesblacklist_optional_multiple_repo_files(current_actor_context):
-    repo_facts = get_repo_facts(['test', 'rhel-7-server-optional-rpms', 'rhel-7-server-rpms'], True, True)
-    current_actor_context.feed(repo_facts)
-    current_actor_context.run()
-    assert not current_actor_context.consume(RepositoriesBlacklisted)
-
-
-def test_repositoriesblacklist_no_optional_multiple_repo_files(current_actor_context):
-    repo_facts = get_repo_facts(['rhel-7-server-rpms', 'rhel-7-extras-rpms'], True, True)
-    current_actor_context.feed(repo_facts)
-    current_actor_context.run()
-    assert current_actor_context.consume(RepositoriesBlacklisted)
-    repoids = current_actor_context.consume(RepositoriesBlacklisted)[0].repoids
-    assert 'codeready-builder-for-rhel-8-x86_64-rpms' in repoids
+def test_with_optionals(monkeypatch):
+    def repositories_mock(*model):
+        mapping = [
+            RepositoryMap(
+                from_id='rhel-7-optional-rpms',
+                to_id='rhel-8-optional-rpms',
+                from_minor_version='all',
+                to_minor_version='all',
+                arch='x86_64',
+                repo_type='rpm',
+            ),
+            RepositoryMap(
+                from_id='rhel-7-blacklist-rpms',
+                to_id='rhel-8-blacklist-rpms',
+                from_minor_version='all',
+                to_minor_version='all',
+                arch='x86_64',
+                repo_type='rpm',
+            ),
+        ]
+        yield RepositoriesMap(repositories=mapping)
+
+    monkeypatch.setattr(api, "consume", repositories_mock)
+    optionals = library._get_list_of_optional_repos()
+    assert len(optionals) == 1
```

Comment:
In this case, you know exactly what's expected output. Please test you received the output you expected instead of the length of the dict.